### DISCUSSION TITLE
Client-side templating?

### DISCUSSION THREAD

**Discussion by stefanuddenberg** [jspsych/jsPsych#637] [Q&A]
I couldn't find anything in the documentation about using client-side HTML templates. As a result, I went ahead and made a `render-template` plugin based off of the `external-html` plugin. It works just like the `external-html` plugin, except that it takes in a `render_data` object that is fed to Mustache to be rendered. This suits my needs just fine, but was there an easier way for me to do this?

**Comment by jodeleeuw** [comment]
Currently there's no built-in template engine, so this is probably the right strategy to take.
